web/client/components/catalog/CatalogServiceEditor.jsx

```diff
@@ -17,7 +17,20 @@ import Message from "../I18N/Message";
 import AdvancedSettings from './editor/AdvancedSettings';
 import MainForm from './editor/MainForm';
 
-export default ({
+const withAbort = (Component) => {
+    return (props) => {
+        const [abortController, setAbortController] = useState(null);
```

Comment:
Kindly let me know if you want me to use a variable with getter setter funcs (utils) instead to register controller rather than a state